rest/replicator_test.go

```diff
@@ -3810,3 +3810,236 @@ func TestActiveReplicatorPullConflictReadWriteIntlProps(t *testing.T) {
+			bodyValue:         "EADGBE",
+			tombstoneExpected: "6-6e092cbb01fd955ffa78b518ecfd3eb2",
+		},
+		{
+			// Revision tie with local digest is higher than the remote digest.
+			// local generation = remote generation:
+			//	- e.g. local is 3-c(T), remote is 3-b
+			name:              "revGenTieLocalDigestHigher",
+			revGenTie:         true,
+			bodyValue:         "qux",
+			tombstoneExpected: "6-6a6d63d3fe4633da279ff316f19c6834",
+		},
+		{
+			// Local revision generation is lower than remote revision generation.
+			// local generation < remote generation:
+			//  - e.g. local is 3-a(T), remote is 4-b
+			name:              "revGenLocalLower",
+			revGenHigh:        true,
+			tombstoneExpected: "6-127a19153ddf2a20df73c9b911ee88c2",
+		},
+		{
+			// Local revision generation is higher than remote revision generation.
+			// local generation > remote generation:
+			//	- e.g. local is 3-a(T), remote is 2-b
+			name:              "revGenLocalHigher",
+			tombstoneExpected: "6-bca6950afb1d225b4f76a07d297eef70",
+		},
+	}
+
+	for _, test := range defaultConflictResolverWithTombstoneTests {
+		t.Run(test.name, func(tt *testing.T) {
+			// Passive
+			rt2 := NewRestTester(t, &RestTesterConfig{
+				TestBucket: base.GetTestBucket(t),
+				DatabaseConfig: &DbConfig{
+					Users: map[string]*db.PrincipalConfig{
+						"alice": {
+							Password:         base.StringPtr("pass"),
+							ExplicitChannels: base.SetOf("alice"),
+						},
+					},
+				},
+				noAdminParty: true,
+			})
+			defer rt2.Close()
+
+			// Make rt2 listen on an actual HTTP port, so it can receive the blipsync request from rt1
+			srv := httptest.NewServer(rt2.TestPublicHandler())
+			defer srv.Close()
+
+			// Build passiveDBURL with basic auth creds
+			passiveDBURL, err := url.Parse(srv.URL + "/db")
+			require.NoError(t, err)
+			passiveDBURL.User = url.UserPassword("alice", "pass")
+
+			// Active
+			rt1 := NewRestTester(t, &RestTesterConfig{
+				TestBucket: base.GetTestBucket(t),
+			})
+			defer rt1.Close()
+
+			defaultConflictResolver, err := db.NewCustomConflictResolver(
+				`function(conflict) { return defaultPolicy(conflict); }`)
+			require.NoError(t, err, "Error creating custom conflict resolver")
+
+			config := db.ActiveReplicatorConfig{
+				ID:          t.Name(),
+				Direction:   db.ActiveReplicatorTypePushAndPull,
+				RemoteDBURL: passiveDBURL,
+				ActiveDB: &db.Database{
+					DatabaseContext: rt1.GetDatabase(),
+				},
+				Continuous:           true,
+				ConflictResolverFunc: defaultConflictResolver,
+				ReplicationStatsMap:  base.SyncGatewayStats.NewDBStats(t.Name()).DBReplicatorStats(t.Name()),
+			}
+
+			// Create the first revision of the document on rt1.
+			docID := t.Name() + "foo"
+			rt1RevIDCreated := createOrUpdateDoc(rt1, docID, "", "foo")
+			requireBodyNotEmpty(rt1, docID, rt1RevIDCreated, "foo")
+
+			// Create active replicator and start replication.
+			ar := db.NewActiveReplicator(&config)
+			defer startReplication(ar)()
```

Comment:
We're manually stopping this replication on line 3993, so we don't need to also stop it in a defer.